citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/AuthorityServiceTest.java

```diff
@@ -238,11 +309,281 @@ public void test_save() {
+                "Rose", "Jane", paging)).thenReturn(entriesJane);
+
+        List<IAuthorityEntry> searchFullNameJane = managerToTest.findByName(user, "Rose", "Jane", page, pageSize);
+
+        Assert.assertFalse(searchFullNameJane.isEmpty());
+        Assert.assertTrue(searchFullNameJane.get(0) != null && searchFullNameJane.size() == 1);
+        Assert.assertTrue(searchFullNameJane.get(0).getName().contains("Jane"));
+        Assert.assertFalse(searchFullNameJane.get(0).getName().contains("Albert"));
+
+    }
+
+    @Test
+    public void test_findByName_withInterchangedFirstNameAndLastName() {
+
+        // Testing authority search by full name with first and last name interchanged
+        List<IAuthorityEntry> searchFullNameReverseOrder = managerToTest.findByName(user, "Albert", "Einstein", page,
+                pageSize);
+
+        for (IAuthorityEntry entry : searchFullNameReverseOrder) {
+            Assert.assertTrue(entry.getName().contains("Einstein") && entry.getName().contains("Albert"));
+        }
+
+    }
+
+    @Test
+    public void test_findByName_resultsNotFound() {
+
+        Mockito.when(authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(),
+                "Peter", "Doe", paging)).thenReturn(new ArrayList<>());
+
+        // Testing authority search by full name, name not found
+        List<IAuthorityEntry> actual3 = managerToTest.findByName(user, "Peter", "Doe", page, pageSize);
+
+        Assert.assertTrue(actual3.isEmpty());
+    }
+
+    @Test
+    public void test_getTotalUserAuthoritiesPages() {
+
+        // testing total number of user authority pages when search authority has last
+        // name as "Albert" and pageSize is 10
+        Mockito.when(authorityRepository
+                .countByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), "", "Albert"))
+                .thenReturn((long) 3);
+        Assert.assertTrue(managerToTest.getTotalUserAuthoritiesPages(user, "", "Albert", pageSize) == 1);
+
+        // testing total number of user authority pages when search authority has first
+        // name as "Jane", last name as "Rose" and pageSize is 10
+        Mockito.when(authorityRepository
+                .countByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), "Jane", "Rose"))
+                .thenReturn((long) 1);
+        Assert.assertTrue(managerToTest.getTotalUserAuthoritiesPages(user, "Jane", "Rose", pageSize) == 1);
+
+        // testing total number of user authority pages when search authority has first
+        // name as "Peter", last name as "Doe" and pageSize is 10
+        Mockito.when(authorityRepository
+                .countByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), "Peter", "Doe"))
+                .thenReturn((long) 0);
+        Assert.assertTrue(managerToTest.getTotalUserAuthoritiesPages(user, "Peter", "Doe", pageSize) == 0);
```

Comment:
unless there is something different about the code that is tested you only need one test for each case (e.g. one for total number is 1, one for total number is 0,e tc). also each of these should be their own test (no results, one result, several results, etc)